lib/nested_errors.rb

```diff
@@ -1,10 +1,27 @@
+# frozen_string_literal: true
+
 module NestedErrors
-  def self.unflatten(hash)
-    hash.each_with_object({}) do |(key, value), all|
-      regex = key.to_s.include?('[') ? /\[|\]\./ : '.'
-      key_parts = key.to_s.split(regex).map!(&:to_sym)
-      leaf = key_parts[0...-1].inject(all) { |h, k| h[k] ||= {} }
-      leaf[key_parts.last] = value
+  def self.unflatten(object, relationships)
+    errors = object.errors.messages.to_hash
+    return errors if relationships.blank?
+
+    relationships.each do |relationship|
```

Comment:
@yunixon you may use `[*relationships]` this will allow to pass `:invoice_items` instead `[:invoice_items]` for single element, which is more readable.